### ISSUE TITLE
ValueError: Expected argument `task` to either be `'binary'`, `'multiclass'` or `'multilabel'` but got 1

### ISSUE THREAD

**Issue by saadtariq-ds** [pyannote/pyannote-audio#1242]
vad_task = VoiceActivityDetection(ami, duration=2., batch_size=128)
vad_model = PyanNet(task=vad_task, sincnet={'stride': 10})
trainer = pl.Trainer(gpus=self.gpus, max_epochs=self.max_epochs)
 trainer.fit(vad_model)

**Comment by waoooNiuBility** [comment]
I've met the question as you too，How did you solve it

**Comment by kaleaniket** [comment]
I'm also getting the same error on Colab, Is their any fix to this?

**Comment by kaleaniket** [comment]
I'm also getting the same error on Colab, Is their any fix to this?

**Comment by saadtariq-ds** [comment]
I solved by downgrading torchmetrics version. Current version `torchmetrics==0.10.3`

**Comment by kaleaniket** [comment]
> I solved by downgrading torchmetrics version. Current version `torchmetrics==0.10.3`

Thanks. It worked for me as well.

**Comment by toto6038** [comment]
So it looks like there's a breaking change on torchmetrics' API (`AUROC.__new__` in `.../miniconda3/envs/pyannote/lib/python3.8/site-packages/torchmetrics/classification/auroc.py`). Downgrading torchmetrics may serve as a workaround, but Pyannote has to adapt to the new API eventually.

**Comment by hbredin** [comment]
> but Pyannote has to adapt to the new API eventually.

Yes, indeed. Feel free to contribute this "adaptation" via a pull request.

**Comment by toto6038** [comment]
> > but Pyannote has to adapt to the new API eventually.
> 
> Yes, indeed. Feel free to contribute this "adaptation" via a pull request.

I'll do it

**Comment by saadtariq-ds** [comment]
Thanks everyone

**Comment by hbredin** [comment]
Re-opening the issue until @toto6038's PR fixes it.
Thanks @toto6038 ;-)

**Comment by toto6038** [comment]
# TL;DR
A release of minor update is enough to avoid the problem as the solution has been done already.

## Detail
After some investigation to the source code, it looks like the existing code has already mitigated the problem and therefore there will be no PR. To better explain the cause of problem, we first examine the version of Pyannote that users get from pip:
https://github.com/pyannote/pyannote-audio/blob/7ead17e86aba5459cef074c126bfd69ce7a230f9/pyannote/audio/tasks/segmentation/mixins.py#L126-L132

It's the change of the interface of AUROC's `__new__()` that results in the ValueError issue.

The latest code in develop branch has done similar work as that in [torchmetrics](https://github.com/Lightning-AI/metrics/blob/7c885d0a2ba8d7e66a4b13ae9ad0dc826d70d041/src/torchmetrics/classification/auroc.py#L477-L484) to return different object depending on the type of the task.

https://github.com/pyannote/pyannote-audio/blob/b56add2e2033060dca28be71563dab5ff0fd3e64/pyannote/audio/tasks/segmentation/mixins.py#L143-L152

**Comment by hbredin** [comment]
Oh! Thanks @toto6038 :-)

To anyone encountering the same issue, and until this fix has been released, please install `pyannote.audio` from `develop` branch (where the issue has been fixed already).
